redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -4,29 +4,48 @@ import scala.collection.compat.immutable.LazyList
 
 import zio._
 import zio.redis.RedisError.ProtocolError
+import zio.redis.RespValue.BulkString
 import zio.stm.{ STM, TMap, TRef, USTM }
 
 private[redis] final class TestExecutor private (
   sets: TMap[String, Set[String]],
   strings: TMap[String, String],
   randomPick: Int => USTM[Int]
 ) extends RedisExecutor.Service {
+
   def execute(command: Chunk[RespValue.BulkString]): IO[RedisError, RespValue] =
     for {
       name   <- ZIO.fromOption(command.headOption).orElseFail(ProtocolError("Malformed command."))
       result <- runCommand(name.asString, command.tail).commit
     } yield result
 
-  private[this] def runCommand(name: String, input: Chunk[RespValue.BulkString]): STM[RedisError, RespValue] =
+  private def errResponse(cmd: String): RespValue.BulkString =
+    RespValue.bulkString(s"(error) ERR wrong number of arguments for '$cmd' command")
+
+  private def succeedNowMayEmpty(emptyReturn: => BulkString, nonEmptyReturn: => BulkString)(implicit
```

Comment:
Let's make it a bit more API specific, and drop the implicit, so:

```suggestion
private def onConnection(command: String, input: Chunk[RespValue.BulkString])(res: => RespValue.BulkString) = STM.succeedNow(if (input.isEmpty) errResponse(command) else res)
```
 
Then you can propagate it to the call site. Alrernatively, you could make it curried in a way that it returna a function from input to result.